[futures.state] Turn note into example (#7027)
cplusplus/draft: source/threads.tex

state.
The result of a shared state is set by
respective functions on the asynchronous provider.
\begin{example}

> Examples for asynchronous providers include promises or tasks.

Reviewer: ```suggestion
Promises and tasks are examples of asynchronous providers.
```
